packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -69,13 +73,25 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
     ///         message.
     uint240 internal msgNonce;
 
+    /// @notice Emitted whenever a message is sent to a destination
+    /// @param destination  Chain ID of the destination chain.
+    /// @param target       Target contract or wallet address.
+    /// @param messageNonce Nonce associated with the messsage sent
+    /// @param sender       Address initiating this message call
+    /// @param message      Message payload to call target with.
+    event SentMessage(uint256 indexed destination, address indexed target, uint256 indexed messageNonce, address sender, bytes message);
```

Comment:
Since we're out of indexed params and sender might be nice to index, one option is include the sender's address in the `messageNonce`, so instead of `messageNonce` just being `version || nonce` (`||` is concatenation)  it becomes `version || sender || nonce`. With a 16 bit version and 160 bit address, that still leaves 80 bits for the nonce. 80 bits seems big enough, but if we're worried it's too small we could always make `msgNonce` a mapping from sender address to next nonce for that sender

Reply:
by embedding the sender this way, how would it be used when trying to use `eth_getFilterLogs`? afaik it wouldn't be possible to do a filterLog query to get all events for a given sender since the nonce would affect the topic value

Reply:
Can you filter on ranges of values for a topic? e.g. if the sender was first you could filter based on the size of the uint, i.e. `>= 0x[address]000...000 && <= 0x[address]fff...fff`. Probably over-engineering and not worth it though

Reply:
hmm yea good point, let me look into if ranges are possible.

Reply:
Looked into this and sadly ranges are not supported